Content.Server/_Impstation/Radio/SelfHeadsetSystem.cs

```diff
@@ -0,0 +1,81 @@
+using Content.Server.Emp;
+using Content.Server._Impstation.Radio.Components;
+using Content.Server.Radio.Components;
+using Content.Shared.Radio;
+using Content.Shared.Radio.Components;
+
+namespace Content.Server._Impstation.Radio;
+
+public sealed class SelfHeadsetSystem : EntitySystem
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SelfHeadsetComponent, EncryptionChannelsChangedEvent>(OnKeysChanged);
+        SubscribeLocalEvent<SelfHeadsetComponent, EmpPulseEvent>(OnEmpPulse);
+        SubscribeLocalEvent<SelfHeadsetComponent, ComponentStartup>(OnAdd);
+
+    }
+
+    /// <summary>
+    /// Used to give an entity access to radio channels when an Encryption Key is inserted, without the need for a headset.
+    /// </summary>
+
+    private void OnKeysChanged(EntityUid uid, SelfHeadsetComponent component, ref EncryptionChannelsChangedEvent args)
```

Comment:
`Entity<Component> ent` is preferred over `EntityUid uid, Component component` - in this case, it would be `private void OnKeysChanged(Entity<SelfHeadsetComponent> ent, ref EncryptionChannelsChangedEvent args)`

other methods should be adjusted to reflect this, too